src/node/mini_miner.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_MINI_MINER_H
+#define BITCOIN_NODE_MINI_MINER_H
+
+#include <txmempool.h>
+
+#include <memory>
+#include <optional>
+#include <stdint.h>
+
+namespace node {
+
+// Container for tracking updates to ancestor feerate as we include ancestors in the "block"
+struct MockMempoolEntry {
+    CAmount fee_individual;
+    CAmount fee_with_ancestors;
+    int64_t vsize_individual;
+    int64_t vsize_with_ancestors;
+    const CTransaction& tx;
+
+public:
```

Comment:
All struct members are public by default, so can either drop `public` here or make the struct a class if the data members above are intended to be private.